### PR TITLE
Multi site install rm keys for new site that wasn't cloned

### LINKED ISSUE #5066: Error with multisite upgrade v6.0.0-v6.1.0
### Describe the bug

Decryption and Key creation issues (Blank Screen) with subsequent site upgrade in multisite

### To Reproduce
Upgrade an additional site after upgrading default.
This problem comes up with the newly upgraded site, which causes a blank screen upon login in if the site's database has data. The problem is similar to the one in this forum issue.

https://community.open-emr.org/t/decryption-and-key-creation-issues-blank-screen-is-not-file-permissions/16059

Deleting sites/siteid/documents/logs_and_misc/methods/sixa and sixb then clearing browser fixes the problem.

I understand why this occurs, one solution would be to create a site-template directory elsewhere (in a location such as src) which contains the contents of the current default directory without the sixa and sixb files. From there, the sites directory can be populated when the site is created via the setup process.

### Client configuration

- Browser: Firefox
- OpenEMR version: 6.1.0
- Operating system: windows 10

### LINKED ISSUE #5082: site created from multisite needs own keys unless site was a clone
### Describe the bug

Decryption and Key creation issues (Blank Screen) with subsequent site creation.

### To Reproduce

Create an additional site after Default.
This problem comes up with the newly created site, which causes a blank screen on going to login screen.

### PR DESCRIPTION
**Pull request by stephenwaite**
Fixes #5082 , thanks @gutiersa

Short description of what this resolves:

removes source keys from new site unless site was a clone

the keys for the new site are created on first login
Changes proposed in this pull request:

### CODE REVIEW

File: library/classes/Installer.class.php [hunk 1]
```diff
@@ -472,6 +472,8 @@ public function create_site_directory()
                 $this->error_message = "unable to copy directory: '$source_directory' to '$destination_directory'. " . $this->error_message;
                 return false;
             }
+            // the new site will create it's own keys so okay to delete these copied from the source site
+            array_map('unlink', glob($destination_directory . "/documents/logs_and_misc/methods/*"));
```
Comment (bradymiller) [inline library/classes/Installer.class.php:476]: Two scenarios.
If cloning the database, then want to keep the keys.
If not cloning the database, then should remove the keys.

Comment (bradymiller) [reply library/classes/Installer.class.php:476]: image

image

**Comment by bradymiller** [comment]
In docker, whenever running setup.php, generally need to always put the `%` for the `User Hostname`. Is this the issue that you noted regarding the `%`?

**Comment by gutiersa** [comment]
In my case I enter localhost. I am not allowed to leave it blank, but I am
not yet using docker.

*Sandragutierrezmd*
*THINK BEFORE YOU PRINT*
This message, together with any attachments, is intended only for the
addressee. It may contain information that is legally privileged,
confidential, and exempt from disclosure. If you are not the intended
recipient, you are hereby notified that any disclosure, copying,
distribution, use, or any action or reliance on this communication is
strictly prohibited. If you have received this e-mail in error, please
notify the sender immediately by telephone, fax, or return e-mail and
delete the message, along with any attachments.

**Comment by stephenwaite** [comment]
> n docker environments, whenever running setup.php, generally need to always put the `%` for the `User Hostname`. Is this the issue that you noted regarding the `%`?

only time it's needed is with multisite I believe but will check out a vanilla install with `openemr-cmd dr`

**Comment by stephenwaite** [comment]
yep, needs `%` for basic docker install as well, ty Brady

**Comment by stephenwaite** [comment]
testing well

**Review by bradymiller — APPROVED** [review]

**Comment by bradymiller** [comment]
118382341-53d9d600-b5c2-11eb-8b7e-220bbab8468a

**Pull request merged**